### PR TITLE
Allows optional axis argument in Dask Array squeeze method

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/dask/dask/issues/3259

### CODE REVIEW

**Comment by jakirkham** [comment]
May need a changelog entry. If so, will handle after PR ( https://github.com/dask/dask/pull/3260 ) is merged.

File: dask/array/routines.py [hunk 1]
```diff
@@ -836,20 +835,26 @@ def ravel(array):
 
 @wraps(np.squeeze)
 def squeeze(a, axis=None):
-    if 1 not in a.shape:
-        return a
     if axis is None:
         axis = tuple(i for i, d in enumerate(a.shape) if d == 1)
-    b = a.map_blocks(partial(np.squeeze, axis=axis), dtype=a.dtype)
-    chunks = tuple(bd for bd in b.chunks if bd != (1,))
+    else:
```
Comment (mrocklin) [inline dask/array/routines.py]: Do we need this `else` here?

**Comment by mrocklin** [comment]
Small comment, otherwise seems good to me

File: dask/array/routines.py (hunk 1 above)
Comment (jakirkham) [reply dask/array/routines.py]: Yes, this handles both `int` and `tuple`s of `int`s. It coerces the former into the latter first (maybe part of the confusion). Then applies some checks and normalizes the input data. None of these checks or normalization are required for `None` as the data is already constructed to meet our expectations.

Comment (mrocklin) [reply dask/array/routines.py]: I agree that the body of the else is useful.  I think we should the normalization code regardless, just to reduce the code nesting.

Comment (jakirkham) [reply dask/array/routines.py]: Sure, fixed. :)

**Comment by jakirkham** [comment]
Alright, think this is ready to go in now. :)

**Comment by jakirkham** [comment]
Forgot a changelog entry. Adding in PR ( https://github.com/dask/dask/pull/3321 ).

**Pull request merged**